lib/clearance/user.rb

```diff
@@ -75,13 +75,13 @@ def reset_remember_token!
     def update_password(new_password)
       self.password_changing = true
       self.password = new_password
+      valid?
 
-      if valid?
+      if errors[:password].blank?
```

Comment:
Some quick testing shows that `empty?` really should be safe though. Even looking for non-existent attributes still returns an empty array.